### ISSUE TITLE
Maximum recursion limit exceeded when running protocol (Trac #38)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#42]
Running a protocol with 16 steps, repeated 100 times results the following:

RuntimeError: maximum recursion depth exceeded

Migrated from http://microfluidics.utoronto.ca/ticket/38
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Running a protocol with 16 steps, repeated 100 times results the following:\n\nRuntimeError: maximum recursion depth exceeded", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Maximum recursion limit exceeded when running protocol", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-01-24T20:04:35", 
    "milestone": "Microdrop 1.0", 
    "owner": "ryan", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120125T20:59:29: 
* ryan@fobel.net commented:

> commit 1fc554da0d0b46a2a5381ff4be1c7d8a55b514d2
> Author: Ryan Fobel <ryan@fobel.net>
> Date:   Wed Jan 25 15:47:31 2012 -0500
> 
>     Replace recursive call during protocol run. Fixes #42.


* ryan@fobel.net changed resolution from "" to "fixed"
* ryan@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
